alacritty/src/renderer/mod.rs

```diff
@@ -1007,16 +1015,22 @@ impl<'a, C> RenderApi<'a, C> {
 
     #[inline]
     fn add_render_item(&mut self, cell: RenderableCell, glyph: &Glyph) {
+        let layer_id =
+            if glyph.colored { LayerType::ColoredGlyphs } else { LayerType::RegularGlyphs };
+        let batch = &self.layers[layer_id as usize].batch;
+
         // Flush batch if tex changing.
-        if !self.batch.is_empty() && self.batch.tex != glyph.tex_id {
-            self.render_batch();
+        if !batch.is_empty() && batch.tex != glyph.tex_id {
+            self.render_batch(layer_id);
         }
 
-        self.batch.add_item(cell, glyph);
+        let batch = &mut self.layers[layer_id as usize].batch;
```

Comment:
This seems a bit unfortunate? Are you accessing things this way because self.render_batch complains when batch is made a mutable reference in the first place?